### PR TITLE
Implement Anger Shell and Berserk

### PR DESCRIPTION
**Pull request by madibye**
Each time your mon drops below half HP, the stat effect activates, giving them either +1 Atk/SpA/Spe and -1 Def/SpD for Anger Shell, or +1 SpA for Berserk.

This is my first time contributing, so lemme know if I fucked up anything :)

### CODE REVIEW

File: src/data/ability.ts [hunk 1]
```diff
@@ -3091,7 +3120,7 @@ export function initAbilities() {
     new Ability(Abilities.STEELWORKER, 7)
       .attr(MoveTypePowerBoostAbAttr, Type.STEEL),
     new Ability(Abilities.BERSERK, 7)
-      .unimplemented(),
+      .attr(PostDefendHpGatedStatChangeAbAttr, (target, user, move) => move.category !== MoveCategory.STATUS, 0.5, BattleStat.SPD, 1),
```
Comment (lucfd) [inline src/data/ability.ts:3123]: Berserk should raise SPATK rather than SPD

**Review by lucfd — CHANGES_REQUESTED** [review]

File: src/data/ability.ts (hunk 1 above)
Comment (madibye) [reply src/data/ability.ts]: Realized that when I went to test it just now >< fixed in 2da59ae

**Comment by lucfd** [comment]
LGTM!

**Review by lucfd — APPROVED** [review]

**Review by Madmadness65 — CHANGES_REQUESTED** [review]

File: src/data/ability.ts [hunk 2]
```diff
@@ -580,6 +580,35 @@ export class PostDefendStatChangeAbAttr extends PostDefendAbAttr {
...
+    this.stat = stat;
+    this.levels = levels;
+    this.selfTarget = selfTarget;
+  }
+
+  applyPostDefend(pokemon: Pokemon, passive: boolean, attacker: Pokemon, move: PokemonMove, hitResult: HitResult, args: any[]): boolean {
+    const hpGateFlat: integer = Math.ceil(pokemon.getMaxHp() * this.hpGate)
+    const lastAttackReceived = pokemon.turnData.attacksReceived[pokemon.turnData.attacksReceived.length - 1]
+    if (this.condition(pokemon, attacker, move.getMove()) && (pokemon.hp <= hpGateFlat && (pokemon.hp + lastAttackReceived.damage) > hpGateFlat)) {
+      pokemon.scene.unshiftPhase(new StatChangePhase(pokemon.scene, (this.selfTarget ? pokemon : attacker).getBattlerIndex(), true, [ this.stat ], this.levels));
```
[the remark this answers is not in the document]
Comment (madibye) [reply src/data/ability.ts]: Done 041b713

**Review by Madmadness65 — APPROVED** [review]
LGTM now

**Pull request merged**